File: C/maximum_average_subarray_I.c
int sum(int* nums, int n){
    int res = 0;
    for(int i = 0; i < n; i++) res += *(nums+i);
    return res;
}

double findMaxAverage(int* nums, int numsSize, int k){
    double s = 0, tmp = 0;
        
    for(int i = 0; i < numsSize-k; i++) {
        tmp += nums[i+k] - nums[i];
        if(tmp > s) s = tmp;
    }
            
    return ((double)sum(nums, k)+s) / k;
}

File: Python3/maximum_average_subarray_I.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        s = tmp = 0
        
        for i in range(len(nums)-k):
            tmp += nums[i+k] - nums[i]
            if tmp > s: s = tmp
            
        return (sum(nums[:k])+s) / k
